XiEditor/EditViewController.swift

```diff
@@ -14,32 +14,86 @@
 
 import Cocoa
 
-class EditViewController: NSViewController {
+/// The EditViewDataSource protocol describes the properties that an editView uses to determine how to render its contents.
+protocol EditViewDataSource {
+    var lines: LineCache { get }
+    var styleMap: StyleMap { get }
+    var textMetrics: TextDrawingMetrics { get }
+    var gutterWidth: CGFloat { get }
+}
 
-    @IBOutlet var editView: EditView!
+
+class EditViewController: NSViewController, EditViewDataSource {
+
+    
     @IBOutlet var shadowView: ShadowView!
     @IBOutlet var scrollView: NSScrollView!
-    @IBOutlet var documentView: NSClipView!
+    @IBOutlet weak var editViewContainer: EditViewContainer!
+    @IBOutlet var editView: EditView!
+    @IBOutlet weak var gutterView: GutterView!
+    
+    @IBOutlet weak var gutterViewWidth: NSLayoutConstraint!
+    @IBOutlet weak var editViewHeight: NSLayoutConstraint!
     
     var document: Document! {
         didSet {
             editView.document = document
         }
     }
+    
+    var lines: LineCache = LineCache()
+    var styleMap: StyleMap {
+        return (NSApplication.shared().delegate as! AppDelegate).styleMap
+    }
+
+    //TODO: preferred font should be a user preference
+    var textMetrics = TextDrawingMetrics(font: CTFontCreateWithName("InconsolataGo" as CFString?, 14, nil))
+    
+    var gutterWidth: CGFloat {
+        return gutterViewWidth.constant
+    }
+    
+    // used to calculate the gutter width. Initial -1 so that a new document
+    // still triggers update of gutter width.
+    private var lineCount: Int = -1 {
+        didSet {
+            if lineCount != oldValue {
+                let gutterColumns = "\(lineCount)".characters.count
```

Comment:
Also isn't the maximum line value already conveyed to the front-end? we use it to determine the height of the document view.